Frontend/HsTypeChecker.hs

```diff
@@ -638,6 +641,8 @@ unify  untchs eqs
       = Just (mempty, [ty1 :~: ty3, ty2 :~: ty4])
```

Comment:
You forgot a case (an important one): 
```haskell
one_step _us (TyUnit :~: TyUnit)
  = Just (mempty, [])
```

Reply:
Derp, consider it done

Reply:
Do me a favor and move the equation to line 642 (before all the `Nothing` / failing cases); feel free to merge otherwise :slightly_smiling_face: